### ISSUE TITLE
Consider traci.vehicle.openGap() method

### ISSUE THREAD

**Issue by leoluecken** [eclipse/sumo#4172]
To create gaps for letting other vehicles join is a common task in cooperative maneuvering, whose implementation in TraCI-based applications could be supported by such a method.

Proposed signature: `createGap(desiredGap, desiredSpeedDiff, maximumDecel, duration)`

`desiredGap`: the space headway, which shall be established by the vehicle

`desiredSpeedDiff`: The difference to the leading vehicle's speed at which the gap shall be established, i.e. the growth rate `v_leader-v_ego>0`, for the gap

`maximumDecel`: the maximal deceleration, that the vehicle should apply to establish the desired speed difference

`duration`: The duration for which the advice shall be active

**Comment by namdre** [comment]
maybe this should be part of simpla

**Comment by leoluecken** [comment]
Within simpla some similar mechanisms exist (to enlarge a gap to a size that allows  safe vType switching), but it is difficult to see how this could be imported into other scripts, such that it remains easy to use.
Simpla on the other hand could make use of such a TraCI function.

Do you have any aversions against incliding it in SUMO directly?

I think it has a lot of similarities with slowDown()...

**Comment by namdre** [comment]
Aversion is too strong a word. 
It's just that this gap creation is a complex manoevre that could be recreated with more basic traci commands. I conjecture that there is a large number of complex maneuvers and we do not want to support them all as top-level traci calls.
Adding them to a client side library does not raise the problem of unchecked API growth and this was my reason for suggesting simpla. 
I realise that the client side library solution falls short in terms of language portability and maybe also in ease of use when using the functionality in other scripts.

In conclusion: if adding this to TraCI directly helps you, then please go ahead and implement it. It won't hurt.

**Comment by leoluecken** [comment]
A Possible inclusion to simpla would need to create an object associated to the vehicle, because we need a state to track the maneuver progress, say a "simpla.GapCreator". It would need to get active every simstep and obtain the required information about the vehicle state. It could be a stepListener in itself.

**Comment by leoluecken** [comment]
Already implemented, see 2df14eb0d44b9e522e7d6cbcc5dddf36510f4f4d

**Comment by leoluecken** [comment]
Completed in a12569b78cc5f17a7cd5b6843699b867a23ef81c

**Comment by leoluecken** [comment]
This came up again in the context of using iTETRIS, where simpla is not accessible (since the iTETRIS uses the C++ interface).

I will implement a solution along the lines of slowDown() as creating a specific gap seems to be a pretty common task in control algorithms and may be of use to many people. I'll follow an approach to increase tau from the current value to a desired one gradually.

Requirements: 
* Allow to limit the braking rate imposed from the increase of tau.
* Specify the rate for the interpolation from the initial to the new value of tau.
* Specify the duration for which the new tau should be active (after the interpolation).
* Switch back to the original tau after the specified duration.
* Allow to interrupt/deactivate the process.

**Comment by leoluecken** [comment]
First working version done.
